src/engraving/layout/layoutsystem.cpp

```diff
@@ -1219,6 +1227,43 @@ void LayoutSystem::processLines(System* system, std::vector<Spanner*> lines, boo
         }
     }
 
+    if (slurSegments.size() > 1) {
+        const qreal slurCollisionVertOffset = 0.65 * system->spatium();
+        const qreal fuzzyHorizCompare = 0.1 * system->spatium();
+        auto compare = [fuzzyHorizCompare](qreal x1, qreal x2) { return qAbs(x1 - x2) < fuzzyHorizCompare; };
+        for (SlurSegment* slur1 : slurSegments) {
+            for (SlurSegment* slur2 : slurSegments) {
+                // slurs don't collide with themselves
```

Comment:
this one looks over complicated and not performant

in general, I think you need to compare only adjacent elements by x point. So that it makes sense to prepare a std::map<int /*x*/, SlurSegment*> (which is sorted by default) and then compare adjacent elements

Reply:
I agree that this is cumbersome, but in this loop I test both endpoints, so there could be slurs which are non-adjacent in terms of start point, but which end on the same chordrest:
![image](https://user-images.githubusercontent.com/89263931/158029442-d8965089-9958-47a0-b259-bf136ecf0fdb.png)
So to test the endpoint collisions every slur segment needs to be compared to every other slur segment as far as I can tell. If there is a better way to do that, please let me know!